Rakefile

```diff
@@ -318,13 +330,17 @@ if false
 
 else # original
 
+  naked_mirahc_jar = new_jar.gsub(".jar","-naked.jar")
 
-  mirah_srcs = Dir['src/org/mirah/{builtins,jvm/types,macros,util,}/*.mirah'].sort +
+  mirah_srcs = Dir['src/org/mirah/{jvm/types,macros,util,}/*.mirah'].sort +
+               Dir['src/org/mirah/builtins/builtins.mirah'].sort +
                Dir['src/org/mirah/typer/**/*.mirah'].sort +
                Dir['src/org/mirah/jvm/{compiler,mirrors,model}/**/*.mirah'].sort +
                Dir['src/org/mirah/tool/*.mirah'].sort
 
-  file new_jar => mirah_srcs + ['src/org/mirah/ant/compile.mirah'] + [old_jar, 'javalib/asm-5.jar', 'javalib/mirah-parser.jar'] do
+  extensions_srcs = Dir['src/org/mirah/builtins/*_extensions.mirah'].sort
+
+  file new_jar => mirah_srcs + extensions_srcs + ['src/org/mirah/ant/compile.mirah'] + [old_jar, 'javalib/asm-5.jar', 'javalib/mirah-parser.jar'] do
```

Comment:
How about extracting a var for the ant source too. I think that'd make this more clear.